stoobly_agent/app/cli/scaffold_cli.py

```diff
@@ -242,9 +243,13 @@ def update(**kwargs):
     if old_hostname != kwargs['hostname']:
       service_config.hostname = kwargs['hostname']
 
-      # If this is the default proxy_mode, assume it is safe to update to match the new hostname
+      # If this is the default proxy_mode and the origin matches the original hostname, assume it is safe to update with the new hostname
       if service_config.proxy_mode.startswith("reverse:"):
-        service_config.proxy_mode = service_config.proxy_mode.replace(old_hostname, service_config.hostname)
+        old_origin = service_config.proxy_mode.split("reverse:")[1]
```

Comment:
What if we are unable to split by `reverse:`? The length of return result would just be one and using referencing the second element would cause an error to be thrown.

Reply:
I believe that's handled by `if service_config.proxy_mode.startswith("reverse:"):`